x-pack/plugins/ai_infra/product_doc_base/server/services/doc_manager/doc_manager.ts

```diff
@@ -0,0 +1,204 @@
+  scheduleInstallAllTask,
+  scheduleUninstallAllTask,
+  scheduleEnsureUpToDateTask,
+  getTaskStatus,
+  waitUntilTaskCompleted,
+} from '../../tasks';
+import { checkLicense } from './check_license';
+import type {
+  DocumentationManagerAPI,
+  DocGetStatusResponse,
+  DocInstallOptions,
+  DocUninstallOptions,
+  DocUpdateOptions,
+} from './types';
+
+const TEN_MIN_IN_MS = 10 * 60 * 1000;
+
+/**
+ * High-level installation service, handling product documentation
+ * installation as unary operations, abstracting away the fact
+ * that documentation is composed of multiple entities.
+ */
+export class DocumentationManager implements DocumentationManagerAPI {
+  private logger: Logger;
+  private taskManager: TaskManagerStartContract;
+  private licensing: LicensingPluginStart;
+  private docInstallClient: ProductDocInstallClient;
+  private auditService: CoreAuditService;
+
+  constructor({
+    logger,
+    taskManager,
+    licensing,
+    docInstallClient,
+    auditService,
+  }: {
+    logger: Logger;
+    taskManager: TaskManagerStartContract;
+    licensing: LicensingPluginStart;
+    docInstallClient: ProductDocInstallClient;
+    auditService: CoreAuditService;
+  }) {
+    this.logger = logger;
+    this.taskManager = taskManager;
+    this.licensing = licensing;
+    this.docInstallClient = docInstallClient;
+    this.auditService = auditService;
+  }
+
+  async install(options: DocInstallOptions = {}): Promise<void> {
+    const { request, force = false, wait = false } = options;
+
+    const { status } = await this.getStatus();
+    if (!force && status === 'installed') {
+      return;
+    }
+
+    const license = await this.licensing.getLicense();
+    if (!checkLicense(license)) {
+      throw new Error('Elastic documentation requires an enterprise license');
+    }
+
+    const taskId = await scheduleInstallAllTask({
+      taskManager: this.taskManager,
+      logger: this.logger,
+    });
+
+    if (request) {
+      this.auditService.asScoped(request).log({
+        message: `User is requesting installation of product documentation for AI Assistants. Task ID=[${taskId}]`,
+        event: {
+          action: 'product_documentation_create',
+          category: ['database'],
+          type: ['creation'],
+          outcome: 'unknown',
+        },
+      });
+    }
+
+    if (wait) {
+      await waitUntilTaskCompleted({
+        taskManager: this.taskManager,
+        taskId,
+        timeout: TEN_MIN_IN_MS,
+      });
+    }
+  }
+
+  async update(options: DocUpdateOptions = {}): Promise<void> {
+    const { request, wait = false } = options;
+
+    const taskId = await scheduleEnsureUpToDateTask({
+      taskManager: this.taskManager,
+      logger: this.logger,
+    });
+
+    if (request) {
+      this.auditService.asScoped(request).log({
+        message: `User is requesting update of product documentation for AI Assistants. Task ID=[${taskId}]`,
+        event: {
+          action: 'product_documentation_update',
```

Comment:
This action needs to be included in the audit logging docs (`docs/user/security/audit-logging.asciidoc`).

Reply:
Hum, it's already there?

https://github.com/elastic/kibana/blob/27abf468bf205d9666fb83be1abc7a419e43ae27/docs/user/security/audit-logging.asciidoc#L340-L341

Reply:
Hmm, sorry about the noise. For whatever reason the vscode extension was only showing me the one event.